guava-tests/benchmark/com/google/common/collect/MapBenchmark.java

```diff
@@ -213,4 +205,30 @@ public enum Impl {
     }
     return dummy;
   }
+
+  @Benchmark boolean iterateWithEntrySet(int reps) {
+    Map<Element, Element> map = mapToTest;
+
+    boolean dummy = false;
+    for (int i = 0; i < reps; i++) {
+      for (Map.Entry<Element, Element> entry : map.entrySet()) {
+        dummy ^= entry.getKey() != entry.getValue();
+      }
+    }
+    return dummy;
+  }
+
+  @Benchmark boolean iterateWithKeySetAndGet(int reps) {
+    Map<Element, Element> map = mapToTest;
+
+    boolean dummy = false;
+    for (int i = 0; i < reps; i++) {
+      for (Element key : map.keySet()) {
+        Element value = map.get(key);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: Iterating over [map.keySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#keySet--) and calling [map.get(Key)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-) to get the values within the loop is inefficient. 
**Fix**: Iterate over [map.entrySet()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) and access the values using [entry.getValue()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.Entry.html#getValue--) to avoid potential performance penalty due to lookup.